rln/src/pm_tree_adapter.rs

```diff
@@ -16,6 +16,9 @@ const METADATA_KEY: [u8; 8] = *b"metadata";
 
 pub struct PmTree {
     tree: pmtree::MerkleTree<SledDB, PoseidonHash>,
+    /// The indices of leaves which are set into zero upto next_index.
+    /// Set to 0 if the leaf is empty and set to 1 in otherwise.
+    cached_leaves_indices: Vec<u8>,
```

Comment:
u8 probably not the best here, the highest index it can take is then 255, maybe should use usize instead or u32.

Reply:
it's not index itself - its 0 or 1 correspond the index to value 
 
 if we end up storing a value, then of course it will be usize